pkg/replication/replication.go

```diff
@@ -287,50 +336,31 @@ func (r replication) updateOne(node types.NodeName, done chan<- types.NodeName,
 		node,
 		r.manifest,
 	)
-	for err != nil {
-		nodeLogger.WithError(err).Errorln("Could not write intent store")
-		r.errCh <- err
-		time.Sleep(1 * time.Second)
-		_, err = r.store.SetPod(
-			kp.INTENT_TREE,
-			node,
-			r.manifest,
-		)
-	}
-
-	completedChan := make(chan struct{})
-	updateOneQuit := make(chan struct{})
-	defer close(updateOneQuit)
-
-	go func() {
-		defer close(completedChan)
-		r.ensureInReality(node, updateOneQuit, nodeLogger, targetSHA)
-		r.ensureHealthy(node, updateOneQuit, nodeLogger, aggregateHealth)
-	}()
 
-	if r.timeout == NoTimeout {
-		select {
-		case <-completedChan:
-		case <-quitCh:
-			return
-		}
-	} else {
-		// Wait until either completion, timeout, or a quit
+	for err != nil {
 		select {
-		case <-completedChan:
-		case <-time.After(r.timeout):
-			r.timedOutReplicationsMutex.Lock()
-			r.timedOutReplications = append(r.timedOutReplications, node)
-			r.timedOutReplicationsMutex.Unlock()
-			r.logger.Errorf("The host '%v' timed out during replication for pod '%v'", node.String(), r.manifest.ID())
-		case <-quitCh:
-			return
+		case <-r.quitCh:
+			return errQuit
+		case <-timeoutCh:
+			return errTimeout
+		case <-r.replicationCancelledCh:
+			return errCancelled
+		default:
+			nodeLogger.WithError(err).Errorln("Could not write intent store")
+			r.errCh <- err
+			time.Sleep(1 * time.Second)
```

Comment:
I wonder if we should consider using an exponential backoff here, e.g. sleep for twice as long after each failure